test/unit/gemspec_tests.rb

```diff
@@ -0,0 +1,231 @@
+        "mv #{@exp_build_path} #{@exp_pkg_path}"
+      ]
+
+      cmd_str, exitstatus, stdout = subject.run_build_cmd
+
+      assert_equal exp_cmds_run.first,   cmd_str
+      assert_equal @scmd_spy.exitstatus, exitstatus
+      assert_equal @scmd_spy.stdout,     stdout
+      assert_equal exp_cmds_run,         @scmd_spy.cmd_strings_run
+    end
+
+    should "complain if any system cmds are not successful" do
+      @scmd_spy = ScmdSpy.new(1)
+      err = nil
+      begin
+        subject.run_build_cmd
+      rescue StandardError => err
+      end
+
+      assert_kind_of BuildError, err
+      exp = "gem build --verbose #{subject.path}\n#{@scmd_spy.stderr}"
+      assert_equal exp, err.message
+    end
+
+  end
+
+  class RunInstallCmdTests < CmdTests
+    desc "`run_install_cmd`"
+
+    should "run a system cmd to install the gem" do
+      exp_cmds_run = [
+        "gem install #{@exp_pkg_path}"
+      ]
+
+      cmd_str, exitstatus, stdout = subject.run_install_cmd
+
+      assert_equal exp_cmds_run.first,   cmd_str
+      assert_equal @scmd_spy.exitstatus, exitstatus
+      assert_equal @scmd_spy.stdout,     stdout
+      assert_equal exp_cmds_run,         @scmd_spy.cmd_strings_run
+    end
+
+    should "complain if the system cmd is not successful" do
+      @scmd_spy = ScmdSpy.new(1)
+      err = nil
+      begin
+        subject.run_install_cmd
+      rescue StandardError => err
+      end
+
+      assert_kind_of InstallError, err
+      exp = "gem install #{@exp_pkg_path}\n#{@scmd_spy.stderr}"
+      assert_equal exp, err.message
+    end
+
+  end
+
+  class RunPushCmdTests < CmdTests
+    desc "`run_push_cmd`"
+
+    should "run a system cmd to push the gem to the push host" do
+      exp_cmds_run = [
+        "gem push #{@exp_pkg_path} --host #{subject.push_host}"
+      ]
+
+      cmd_str, exitstatus, stdout = subject.run_push_cmd
+
+      assert_equal exp_cmds_run.first,   cmd_str
+      assert_equal @scmd_spy.exitstatus, exitstatus
+      assert_equal @scmd_spy.stdout,     stdout
+      assert_equal exp_cmds_run,         @scmd_spy.cmd_strings_run
+    end
+
+    should "complain if the system cmd is not successful" do
+      @scmd_spy = ScmdSpy.new(1)
+      err = nil
+      begin
+        subject.run_push_cmd
+      rescue StandardError => err
+      end
+
+      assert_kind_of PushError, err
+      exp = "gem push #{@exp_pkg_path} --host #{subject.push_host}\n#{@scmd_spy.stderr}"
+      assert_equal exp, err.message
+    end
+
+  end
+
+  class ScmdSpy
+
+    attr_reader :cmd_strings_run
+    attr_accessor :cmd_string, :exitstatus
+
+    def initialize(exitstatus)
+      @exitstatus = exitstatus.to_i
+      @cmd_strings_run = []
+    end
+
+    def run
+      @cmd_strings_run << self.cmd_string
+    end
+
+    def success?
+      self.exitstatus == 0
+    end
+
+    def stdout
+      @stdout ||= Factory.text
+    end
+
+    def stderr
+      @stderr ||= Factory.text
+    end
+
+  end
```

Comment:
@kellyredding - Might make a note to put this into scmd. I kinda thought it had one of these already but it doesn't seem like it. I can't remember if we've done a scmd spy in another repo, but it seems familiar.